### PR TITLE
ddl: implement table granularity DDL for SchemaTracker

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/35933

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: ddl/schematracker/checker.go [hunk 1]
```diff
@@ -62,13 +63,32 @@ func (d *Checker) Enable() {
 	d.closed = false
 }
 
+// CreateTestDB creates a `test` database like the default behaviour of TiDB.
+func (d Checker) CreateTestDB() {
```
Comment (Defined2014) [inline ddl/schematracker/checker.go:67]: Could move `CreateTestDB` and `createTestDB` into `NewSchemaTracker` function?

File: ddl/schematracker/checker.go [hunk 2]
```diff
@@ -87,6 +107,46 @@ func (d Checker) checkDBInfo(ctx sessionctx.Context, dbName model.CIStr) {
...
+			return
+		}
+		if tableInfo == nil {
+			errStr := fmt.Sprintf("%s.%s should not exist", dbName, tableName)
+			panic(errStr)
+		}
+		if tableInfo2 == nil {
+			errStr := fmt.Sprintf("%s.%s should exist", dbName, tableName)
+			panic(errStr)
+		}
```
Comment (Defined2014) [inline ddl/schematracker/checker.go:129]: ditto

File: ddl/schematracker/checker.go [hunk 3]
```diff
@@ -62,13 +63,32 @@ func (d *Checker) Enable() {
...
+		if dbInfo == nil && dbInfo2 == nil {
+			return
+		}
+		if dbInfo == nil {
+			errStr := fmt.Sprintf("%s should not exist", dbName)
+			panic(errStr)
+		}
+		if dbInfo2 == nil {
+			errStr := fmt.Sprintf("%s should exist", dbName)
+			panic(errStr)
```
Comment (Defined2014) [inline ddl/schematracker/checker.go:88]: Maybe we could combine those lines like 
```
errStr := fmt.Sprintf("inconsistent dbInfo, dbName: %s, real ddl: %p, schematracker：%p", dbName, dbInfo, dbInfo2)
panic(errStr)
```

I think it's better.

File: ddl/schematracker/checker.go (hunk 1 above)
Comment (lance6716) [reply ddl/schematracker/checker.go:67]: in normal DM using case, `test` database is not wanted.

File: ddl/schematracker/checker.go (hunk 3 above)
Comment (lance6716) [reply ddl/schematracker/checker.go:88]: will do it tomorrow

Comment (lance6716) [reply ddl/schematracker/checker.go]: https://github.com/pingcap/tidb/pull/36077/commits/b57935ebae48c6ffbb9579d0586fe956fc22883c

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**